models/model_choice.py

```diff
@@ -138,7 +136,4 @@ def define_model(
     )
     model = to_dp_model(model=model, devices=devices[1:]) if len(devices) > 1 else model
     model.to(main_device)
-    if state_dict_path:
```

Comment:
In my opinion, the logic here works well, why are we taking it out?

Reply:
With the feature from this PR, loading weights from checkpoint in this function with its current usage would mean:
1. [Reading checkpoint in main(), and optionnaly downloading weights to checkpoint_dir if url](https://github.com/NRCan/geo-deep-learning/pull/457/files#diff-b85c0e821d4ba746c7ae4190dbc4ebafd445d4231a8a1f9796cd238fd79cc052R342)
2. If checkpoint was url, read_checkpoint() would need to return not only the checkpoint dictionary containing params and weights, but also the new path to downloaded local checkpoint (not url) for further use by define_model(). 
3. Define_model() would then take this updated path to read checkpoint a second time into memory, then perform the "model.load_state_dict" operation from those weights. 
Since, all of this would require reading two times rathen than once the provided checkpoint, it wouldn't be very optimal in my opinion (and would also require more lines of code). 

With this in mind, this PR would remove the "model loading" step in define_model() and use define_model() to reallty just "define model's architecture and push it to device"

Reply:
I realize this change might need some discussion and should be tackled by separate PR, prior to this one. @valhassan please review: https://github.com/NRCan/geo-deep-learning/pull/461